File: SpendTrack/Services/CategoryService.cs
using Microsoft.EntityFrameworkCore;
using SpendTrack.Api.Data;
using SpendTrack.Api.Entities;

namespace SpendTrack.Api.Services
{
    public class CategoryService
    {
        private readonly ApplicationDbContext _context;

        public CategoryService(ApplicationDbContext context)
        {
            _context = context;
        }

        public async Task<List<Category>> GetAllCategories()
        {
            return await _context.Categories
                .OrderByDescending(c => c.Id)
                .ToListAsync();
        }

        public async Task<Category> CreateCategory(Category category)
        {
            _context.Categories.Add(category);
            await _context.SaveChangesAsync();
            return category;
        }
    }
}

File: SpendTrack.Web/src/services/categoryService.ts
import api from "./api";
import type { Category } from "../types";

export const getCategories = async (): Promise<Category[]> => {
    const response = await api.get("/category");
    return response.data;
};

export const createCategory = async (category: Omit<Category, "id">): Promise<Category> => {
    const response = await api.post("/category", category);
    return response.data;
};